haystack_experimental/dataclasses/tool.py

```diff
@@ -95,3 +95,16 @@ def from_dict(cls, data: Dict[str, Any]) -> "Tool":
         """
         data["function"] = deserialize_callable(data["function"])
         return cls(**data)
+
+
+def deserialize_tools_inplace(data: Dict[str, Any], key: str = "tools"):
+    """
+    Deserialize tools in a dictionary inplace.
+
+    :param data:
+        The dictionary with the serialized data.
+    :param key:
+        The key in the dictionary where the tools are stored.
+    """
+    if key in data:
+        data[key] = [Tool.from_dict(tool) for tool in data[key]]
```

Comment:
Let's do some type checking with `isinstance` to ensure that we have a list of tools at that key (and raise if we don't).

Reply:
done